### PR TITLE
Pass the build_pk to the task instead of the build object itself

### PR DESCRIPTION
**Pull request by ericholscher**
This breaks in celery when we will ship to prod.

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -1808,13 +1808,14 @@ def retry_domain_verification(domain_pk):
 
 
 @app.task(queue='web')
-def send_build_status(build, state):
+def send_build_status(build_pk, state):
     """
     Send Build Status to Git Status API for project external versions.
 
-    :param build: Build
+    :param build_pk: Buil primary key
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
    :param build_pk: Build primary key
```

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -1834,16 +1835,15 @@ def send_build_status(build, state):
     # TODO: Send build status for other providers.
 
 
-def send_external_build_status(build_pk, state):
+def send_external_build_status(version, build_pk, status):
     """
     Check if build is external and Send Build Status for project external versions.
 
     :param build_pk: Build pk
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
     :param version: Version instance
     :param build_pk: Build pk
```

File: readthedocs/projects/tasks.py [hunk 3]
```diff
@@ -580,12 +580,12 @@ def run_build(self, docker, record):
...
+                version=self.version, build_pk=self.build['id'], status=BUILD_STATUS_FAILURE
             )
         elif self.build_env.successful:
             # send build successful status to git Status API
             send_external_build_status(
-                self.build['id'], BUILD_STATUS_SUCCESS
+                version=self.version, build_pk=self.build['id'], status=BUILD_STATUS_SUCCESS
             )
         else:
             msg = 'Unhandled Build State'
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
             msg = 'Unhandled Build Status'
```

File: readthedocs/projects/tasks.py [hunk 4]
```diff
@@ -1808,13 +1808,14 @@ def retry_domain_verification(domain_pk):
 
 
 @app.task(queue='web')
-def send_build_status(build, state):
+def send_build_status(build_pk, state):
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
def send_build_status(build_pk, status):
```

File: readthedocs/projects/tasks.py [hunk 5]
```diff
@@ -1808,13 +1808,14 @@ def retry_domain_verification(domain_pk):
...
     """
     Send Build Status to Git Status API for project external versions.
 
-    :param build: Build
+    :param build_pk: Buil primary key
     :param state: build state failed, pending, or success to be sent.
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
     :param status: build status failed, pending, or success to be sent.
```

File: readthedocs/projects/tasks.py [hunk 6]
```diff
@@ -1834,16 +1835,15 @@ def send_build_status(build, state):
     # TODO: Send build status for other providers.
 
 
-def send_external_build_status(build_pk, state):
+def send_external_build_status(version, build_pk, status):
     """
     Check if build is external and Send Build Status for project external versions.
 
     :param build_pk: Build pk
     :param state: build state failed, pending, or success to be sent.
```
Comment (saadmk11) [inline readthedocs/projects/tasks.py]:
```suggestion
     :param status: build status failed, pending, or success to be sent.
```

**Comment by ericholscher** [comment]
Yea, we could just send the type along as well. I don't have a strong feeling, but lets merge this so it works, and we can iterate as we go.

**Pull request merged**